hack/generator/pkg/astmodel/storage_conversion_factories.go

```diff
@@ -287,31 +309,42 @@ func assignArrayFromArray(
 
 	srcEp := sourceEndpoint.WithType(srcArray.element)
 	dstEp := destinationEndpoint.WithType(dstArray.element)
-	conversion, _ := createTypeConversion(srcEp, dstEp)
+	conversion, _ := createTypeConversion(srcEp, dstEp, conversionContext)
 
 	if conversion == nil {
 		// No conversion between the elements of the array, so we can't do the conversion
 		return nil
 	}
 
 	return func(reader dst.Expr, writer dst.Expr, generationContext *CodeGenerationContext) []dst.Stmt {
-		itemId := sourceEndpoint.CreateSingularLocal()
-		indexId := itemId + "Index"
-		tempId := sourceEndpoint.CreatePluralLocal("List")
+		// We create three obviously related identifiers to use for the conversion
+		id := sourceEndpoint.CreateLocal()
+		itemId := id + "Item"
```

Comment:
Should these be added to the known locals to avoid subsequent conversions from clobbering them?